libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -133,6 +133,21 @@ string IRGeneratorForStatements::code() const
 	return m_code.str();
 }
 
+void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _varDecl)
+{
+	if (!_varDecl.isConstant() && _varDecl.value())
```

Comment:
reverse condition and bail out early?

Reply:
Actually I think this check should be done at the caller and replaced by an assertion here.

Reply:
At least the non-constant check, the value-check should stay here.